src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -873,6 +873,42 @@ extends scala.collection.concurrent.Map[K, V]
     insertifhc(k, hc, v, INode.KEY_ABSENT)
   }
 
+  // TODO once this is added to concurrent.Map, move the comment there
+  /** If the specified key is not already in the map, computes its value using
+   *  the given mapping function `f` and enters it into the map.
+   *
+   *  Since concurrent maps cannot contain `null` for keys or values,
+   *  a `NullPointerException` is thrown if the given mapping function `f`
+   *  returns `null`.
+   *
+   *  If the specified mapping function throws an exception,
+   *  that exception is rethrown.
+   *  
+   *  Note: this method may invoke the specified function `f` more than once,
+   *  even if the key is in the end not added to the map.
+   *
+   *  @param k      the key to modify
+   *  @param f      the mapping function that computes the value from the key
+   *  @return       the newly added value
+   */
+  def computeIfAbsent(k: K, f: K => V): V = {
```

Comment:
The name may be misleading.  It doesn't just compute; it inserts, also.  However, I wonder why we need this method at all?  The caller has `k` in hand already, and we're not going to use `f` on any other value than `k`, so why not let them do the computation?  (That is, what's wrong with `getOrElseUpdate`, aside from the overly long name?)